mypy/fastparse.py

```diff
@@ -294,11 +294,50 @@ def translate_expr_list(self, l: Sequence[AST]) -> List[Expression]:
             res.append(exp)
         return res
 
-    def translate_stmt_list(self, l: Sequence[AST]) -> List[Statement]:
+    def get_line(self, n: ast3.stmt) -> int:
+        if (isinstance(n, (ast3.AsyncFunctionDef, ast3.ClassDef, ast3.FunctionDef))
+                and n.decorator_list):
+            return n.decorator_list[0].lineno
+        return n.lineno
+
+    def translate_stmt_list(self,
+                            l: Sequence[ast3.stmt],
+                            module: bool = False) -> List[Statement]:
+
+        # A "# type: ignore" comment before the first statement of a module
+        # ignores the whole module:
+
+        if module and l and self.type_ignores and min(self.type_ignores) < self.get_line(l[0]):
+            self.errors.used_ignored_lines[self.errors.file].add(min(self.type_ignores))
+            b = Block(self.fix_function_overloads(self.translate_stmt_list(l)))
+            b.is_unreachable = True
+            return [b]
+
         res = []  # type: List[Statement]
-        for e in l:
+        line = 0
+
+        for i, e in enumerate(l):
+
+            if module:  # and...
```

Comment:
Somehow it looks inefficient to check for `module` and `sys.version_info` each time through the loop. Also the comment here adds nothing. You did know `and` short-circuits, right? You can combine the two checks and save one level of indentation.

Reply:
It was originally `if module and sys.version_info >= (3, 8):`, but the mypy self check complained when I accessed `e.end_lineno` on 336. As I understand it, the mypy's version checks don't work for compound tests like this? Splitting it up silences the error, and I believe the emitted bytecode is identical.

If it would be clearer, I can just `# type: ignore` line 336 instead!

Reply:
OK, then the two `if` statements is optimal -- just put in a clearer comment explaining that this is to work around a mypy issue.